src/components/ui/UiTimeline.vue

```diff
@@ -0,0 +1,97 @@
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue'
+import type { Ref, PropType } from 'vue'
+import type { TimeLineItem } from '../../model/timeLineItems.model'
+
+const props = defineProps({
+  items: {
+    type: Array as PropType<TimeLineItem[]>,
+    default: () => [],
+  },
+})
+
+const emit = defineEmits(['current-date'])
+
+let circle: Ref<HTMLDivElement | null> = ref(null)
+let container: Ref<HTMLDivElement | null> = ref(null)
+let currentActive: Ref<number> = ref(0)
```

Comment:
CMIIW, but is it possible to put this `currentActive` as a prop? If so, then we can bind the value of `currentActive` from the pinia state or computed. If we put `currentActive` in the prop, we can then see it in the histoire
Also, maybe `selectedIndex` is more explanatory than `currentActive` for me. We can compare if the index is selected (e.g. `if (selectedIndex === index)`)